controllers/openstackmachine_controller.go

```diff
@@ -334,11 +335,16 @@ func (r *OpenStackMachineReconciler) reconcileNormal(ctx context.Context, logger
 	case infrav1.InstanceStateActive:
 		logger.Info("Machine instance is ACTIVE", "instance-id", instance.ID)
 		openStackMachine.Status.Ready = true
-	case infrav1.InstanceStateBuilding:
-		logger.Info("Machine instance is BUILDING", "instance-id", instance.ID)
-	default:
+	case infrav1.InstanceStateBuild:
+		logger.Info("Machine instance is BUILD", "instance-id", instance.ID)
+	case infrav1.InstanceStateError:
```

Comment:
case infrav1.InstanceStateError, infrav1.InstanceStateDeleted, infrav1.InstanceStateUnknown:

Because DELETED, UNKNOWN statuses also never become ACTIVE again.

Reply:
UNKNOWN can become ACTIVE. It is returned in certain circumstances if the control plane can't reach the compute, but doesn't mean that the VM has failed. IIRC the primary use cases are Edge and during cell maintainance on multi-cell deployments.

It might also be better not to handle DELETED here, because that will end reconciliation in failure. DELETED presumably means the user deleted the VM for some reason, almost certainly in the hope that it would fix a problem. If we continue to ignore DELETED here we'll automatically create a new machine when the deletion completes, which is most likely the intended behaviour.